### ISSUE TITLE
Feature requests: Support for change logs and semi-automatic mode

### ISSUE THREAD

**Issue by MrUzagi** [AcidWeb/CurseBreaker#112]
### Feature Requests
**(TL;DR at the bottom.)**

I've only used this for a few of weeks now, but I fell in love the first time. It keeps getting better with each update. Heck, you even got a "compact mode" before I got around to suggest it. :-)

That being said. To be able to give Twitch Client the well deserved boot there's a few things I'm still missing.

####  Change Logs
Call me nerdy, but I like to see what changes there have been. If the author went to the effort of including them (not all do, mind) there could be important information in them. So it would be nice to be able to see the change logs for all the installed addons. :-)
Probably not with the usual `status` command, but perhaps a `list` command that simply lists the installed addons without checking for new versions?

Some change logs can be quite wordy (like this feature request I suppose :-) ) so I'm not entirely sure how it would be implemented and still be useful

#### Semi-Automatic Mode
Personally I also like to read change logs (if there are any) _before_ I update. (_Especially around bigger game patches, that frequently break addons. As the game patch lands here in the EU a day later, installing the new addon versions before the game patch actually lands tend to break more things._)

I also like to see what would be updated before I give the go ahead. So I'd like to have a semi-automatic mode when I start the client. Where after the usual five second timeout a status report is printed (with the change logs if available), the execution waits and gives the choice to update or not. There could be an option that if there were no change logs it instead could print out "No change logs found", wait another five seconds and then automatically go ahead with the update anyway. (Would that be semi-semi-automatic? :-D )

#### List Command
With the proposed support for change logs there would be a need for a command to `list` the addons to show the change logs for currently installed addons without having to check for new versions of them every time. This could be implemented in a few ways. With out knowing how the command parser works, I don't know what would work best.

Either there could be a `list` that works like the `status` with the exception of only listing the addons with out checking for new versions. Combined with a `list_changelogs` that also lists change logs (if any.)

Or there could be `list` with various parameters. For example `list addons` to show addons, `list changelogs` to show addons including change logs, or even `list settings`to show the current settings of CurseBreaker.

####Conclusion?
This became a lot more wordy than I anticipated when I started writing this. I didn't set out to write a thesis exactly, so I'm not certain what conclusions we can make from this. I had a few more ideas that popped up while writing, but I better stop while there's a chance anyone will even read this. :-D
Whatever you make of this, and whether you implement any of my suggestions or not, I'm sure CurseBreaker will continue to become better and I'm happy to be along for the ride. Keep it up. :-)

### TL;DR
I propose the following (for more details see above):
1. Add support for showing change logs.
   * For installed addons this could be done with a new `list`command (see *3* below).
   * For updated addons this could be done in a status report before or after the actual update (see *2* below).
2. Add support for a semi-automatic mode that shows what will be installed and waits for user input before continuing. (A `set`or `toggle` command.)
   * Optionally showing change logs. (`toggle_changelogs` ?)
   * Optionally continuing with the install if there are no new change logs.
3. Implement a `list` command works like `status` but that simply lists the installed addons without checking for new versions.
   * Optionally shows change logs. ( `list_changelogs` or `list changelogs` ?)

**Comment by AcidWeb** [comment]
Thank you for feedback. TLDR: No :-D

Changelogs are no-go. WoWI ones are overflowed by bbcode and parsing them to form that don't suck it not worth the time investment. CF ones are not part of main payload and require additional calls to get them plus also require parsing. And additionally in most cases they just contain list of commits and don't provide any real information.
If you really need them - this software will not meet your needs.

As addon author and person who support players that screwed their UIs I consider not updating the addons as anti-pattern. If I could I would force-feed all of you with updates :-) Even If that would create headaches around patch days.

**Comment by MrUzagi** [comment]
Ok, fair enough. Didn't hurt to ask at least. :-)

No, I don't _need_ change logs, but I think they are useful. I can work around it. CurseBreaker is too good to not use anyway.
